lib/fastlane/plugin/wpmreleasetoolkit/actions/common/create_release_backmerge_pull_request_action.rb

```
@@ -0,0 +1,163 @@
+            labels: labels
+          )
+        end
+
+        pr_urls
+      end
+
+      def self.determine_target_branches(release_version, default_branch)
+        release_branches = Actions.sh('git branch -r -l "origin/release/*"').chomp
+
+        all_release_branches_versions = release_branches
+                                        .split("\n")
+                                        .map { |branch| branch.match(%r{origin/release/([0-9.]*)})&.captures&.first }
+                                        .compact
+
+        target_branches = all_release_branches_versions.select { |branch| Gem::Version.new(branch) > Gem::Version.new(release_version) }
+                                                       .map { |v| "release/#{v}" }
+        target_branches = [default_branch] if target_branches.empty?
+
+        target_branches
+      end
+
+      def self.create_backmerge_pr(token:, repository:, title:, head_branch:, base_branch:, milestone:, labels:)
+        intermediate_branch = "merge/#{head_branch.gsub('/', '-')}-into-#{base_branch.gsub('/', '-')}"
+        Fastlane::Helper::GitHelper.create_branch(intermediate_branch)
+
+        other_action.push_to_git_remote(tags: false)
+
+        pr_body = <<~BODY
+          Merging `#{head_branch}` into `#{base_branch}`.
+
+          Via intermediate branch `#{intermediate_branch}`, to help fix conflicts if any:
+          ```
+          #{head_branch.rjust(40)}  ----o-- - - -
+          #{' ' * 40}       \\
+          #{intermediate_branch.rjust(40)}        `---.
+          #{' ' * 40}             \\
+          #{base_branch.rjust(40)}  ------------x- - -
+          ```
+        BODY
+
+        other_action.create_pull_request(
+          api_token: token,
+          repo: repository,
+          title: title,
+          body: pr_body,
+          head: intermediate_branch,
+          base: base_branch,
+          labels: labels,
+          milestone: milestone
+        )
```

We need to protect against the case where there's nothing to merge.

 - We don't have that use case in Tumblr, because even if there hasn't been any new PR on the `release/*` branch by the time we create a new beta and a backmerge PR, we always add one more commit to the backmerge PR before creating it (to restore the version file to its state on the target branch), so our PRs there are never empty.
 - Arguably, for other apps, it's also probably very rare to have an empty PRs, because on those other apps, unlike Tumblr, usually we only trigger a new beta (and thus a backmerge PR) if there has been betafix PRs merged in the `release/*` branch, otherwise we don't do a beta at all anyway.

BUT, there can still be cases where this happen, especially if:
 - Either the fix for the bug was already cherry-picked in the target branch by the developers (i.e. they fixed the bug already both on `release/*` and `trunk` and then triggered a beta)
 - Or for some reason the CI failed mid-flight on the first try doing the beta, and the Release Manager is retrying the CI step. So if on first try the (non-empty) backmerge PR was already created… and then merged, but something else failed later in the lane, on second try the backmerge PR would be empty.

---

We could protect against that by wrapping the `create_pull_request` call within a `begin…rescue…end` for example, though it'd be better to be able to detect an empty diff ourselves, even before creating the intermediate branch and doing the GitHub API call.

---

We might also want to protect against "a PR already exists for this branch" error (especially in case the CI failed on first try but already created the backmerge PR… and the Release Manager is now doing a retry of the lane in ReleaseV2, without having closed or merged the first backmerge PR first)?

Though maybe in those cases it's better for the action to fail explicitly, rather than to have it close the existing PR automatically before creating one anew (or rebasing the existing branch… but that feels even more risky to automate a rebase)? 🤔

---

> We need to protect against the case where there's nothing to merge.

I think even in cases of a cherry-pick, there's always at least a version bump?
The scenario where I see it happening is when we're talking about non-release branches, without a version bump. Still, that seems quite rare. Doesn't hurt to add a check though (as the call to the API will indeed fail) 🤔 

> Though maybe in those cases it's better for the action to fail explicitly

We have this case of the language fetch on Tumblr, where we force-push the new changes + try to create the PR, which I don't think even fails when a PR already exists (I think it just logs an error?).

---

No there's also the case of a retry. e.g. `finalize_release` -> creates the backmerge PR, the RM merges that backmerge PR, but then the trampoline build fails for some reason (e.g. ASC or PSC API outage or similar unrelated to our side of the code or something that would need to be fixed in `buildkite-ci` or somewhere else but would not need any commit on the `release/*` branch of the repo to fix it…), and then the RM retries the `finalize_release`, which will try to create a new backmerge PR… but since the first one has already been created and merged and there hasn't been anything new pushed on `release/*` branch since, then PR would be empty and cause a GitHub API error in return.

---

Added diff validation on 42b2ad44f83de2869e1b41360b21d40891931c55